### PR TITLE
feat(plugin): add onFieldDefinition / onArgDefinition / onInputFieldDefinition

### PR DESCRIPTION
**Pull request by tgriesser**
For #635

### CODE REVIEW

File: tests/plugin.spec.ts [hunk 1]
```diff
@@ -292,6 +295,61 @@ describe('plugin', () => {
...
+              },
+            }),
+          },
+          resolve: () => [true],
+        }),
+      ],
+      plugins: [
+        plugin({
+          name: 'Node',
+          onFieldDefinition(field, config) {
```
Comment (jasonkuhrt) [inline tests/plugin.spec.ts:329]: Could it make sense to call this `onOutputFieldDefinition`?

**Review by jasonkuhrt — APPROVED** [review]
LGTM!

File: tests/plugin.spec.ts (hunk 1 above)
Comment (tgriesser) [reply tests/plugin.spec.ts]: Good call. Done.

**Review by Weakky — APPROVED** [review]
Love the idea, LGTM 👍

**Pull request merged**